### PR TITLE
Implement End Error Event. Resolves #31

### LINKED ISSUE #31: Implement End Error Event

### PR DESCRIPTION
**Pull request by caleeli**

### CODE REVIEW

File: src/ProcessMaker/Nayra/Contracts/Bpmn/ErrorEventDefinitionInterface.php [hunk 1]
```diff
@@ -0,0 +1,31 @@
+<?php
+
+namespace ProcessMaker\Nayra\Contracts\Bpmn;
+
+/**
+ * ConditionalEventDefinition interface.
```
Comment (danloa) [inline src/ProcessMaker/Nayra/Contracts/Bpmn/ErrorEventDefinitionInterface.php]: The doc. comment must be corrected to ErrorEvent definition interface

File: src/ProcessMaker/Nayra/Contracts/Bpmn/ErrorInterface.php [hunk 2]
```diff
@@ -0,0 +1,28 @@
+<?php
+
+namespace ProcessMaker\Nayra\Contracts\Bpmn;
+
+/**
+ * Payload for the ErrorEventDefinition
```
Comment (danloa) [inline src/ProcessMaker/Nayra/Contracts/Bpmn/ErrorInterface.php]: Correct the interface documentation to specify that it is the ErrorInterface and not the payload of error event definition

File: tests/Feature/Engine/ErrorEndEventTest.php [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+ */
+class ErrorEndEventTest extends EngineTestCase
+{
+
+    /**
+     * Test a global Error End Event
+     *
+     */
+    public function testErrorEndEventTopLevel()
+    {return;
```
Comment (danloa) [inline tests/Feature/Engine/ErrorEndEventTest.php]: This function is returning at the beginning, so it is not executed...

File: tests/Feature/Engine/files/Error_EndEvent_TopLevel.bpmn [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+
+        <bpmn2:sequenceFlow id="SequenceFlow_10" name="" sourceRef="end" targetRef="EndEvent_2"/>
+
+        <bpmn2:endEvent id="EndEvent_2">
+            <bpmn2:incoming>SequenceFlow_10</bpmn2:incoming>
+            <bpmn2:errorEventDefinition id="TerminateEventDefinition_1" errorRef="error"/>
+        </bpmn2:endEvent>
+
+    </bpmn2:process>
+</bpmn2:definitions>
```
Comment (danloa) [inline tests/Feature/Engine/files/Error_EndEvent_TopLevel.bpmn]: Add a new line at the file end

File: tests/ProcessMaker/Models/Error.php [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+     * Returns the name of the message
+     *
+     * @return string
+     */
+    public function getName()
+    {
+        return $this->getProperty(ErrorInterface::BPMN_PROPERTY_NAME);
+    }
+
+    public function getErrorCode()
```
Comment (danloa) [inline tests/ProcessMaker/Models/Error.php]: Add the documentation of the method

File: src/ProcessMaker/Nayra/Contracts/Bpmn/ErrorEventDefinitionInterface.php (hunk 1 above)
Comment (caleeli) [reply src/ProcessMaker/Nayra/Contracts/Bpmn/ErrorEventDefinitionInterface.php]: fixed

File: src/ProcessMaker/Nayra/Contracts/Bpmn/ErrorInterface.php (hunk 2 above)
Comment (caleeli) [reply src/ProcessMaker/Nayra/Contracts/Bpmn/ErrorInterface.php]: fixed

File: tests/Feature/Engine/ErrorEndEventTest.php (hunk 3 above)
Comment (caleeli) [reply tests/Feature/Engine/ErrorEndEventTest.php]: Fixed

File: tests/Feature/Engine/files/Error_EndEvent_TopLevel.bpmn (hunk 4 above)
Comment (caleeli) [reply tests/Feature/Engine/files/Error_EndEvent_TopLevel.bpmn]: Fixed

File: tests/ProcessMaker/Models/Error.php (hunk 5 above)
Comment (caleeli) [reply tests/ProcessMaker/Models/Error.php]: Completed

**Pull request merged**